lib/minio/client.rb

```diff
@@ -0,0 +1,190 @@
+    objects.concat(parsed_objects[0])
+
+    is_truncated = parsed_objects[1]
+    continuation_token = parsed_objects[2]
+    while is_truncated
+      query = URI.encode_www_form({
+        "continuation-token" => continuation_token,
+        "delimiter" => "",
+        "encoding-type" => "url",
+        "list-type" => 2,
+        "prefix" => folder_path,
+        "max-keys" => max_keys,
+        "start-after" => continuation_token
+      })
+      response = send_request("GET", s3_uri("#{bucket_name}?#{query}"))
+      parsed_objects = parse_list_objects(response.data[:body])
+      objects.concat(parsed_objects[0])
+      is_truncated = parsed_objects[1]
+      continuation_token = parsed_objects[2]
+    end
+
+    objects
+  end
+
+  def send_request(method, uri, body = nil)
+    headers = @signer.build_headers(method, uri, body, @creds, REGION)
+
+    full_path = uri.path + (uri.query ? "?" + uri.query : "")
+    response = @client.request(method: method, path: full_path, headers: headers, body: body)
+    if [200, 204, 206, 404].include?(response.status)
+      response
+    else
+      raise "Error: #{response.body}"
+    end
+  end
+
+  private
+
+  def parse_list_objects(response)
+    # Parse the XML response
+    doc = Nokogiri::XML(response)
+    bucket_name = doc.xpath("//xmlns:Name").text
+    encoding_type = doc.xpath("//xmlns:EncodingType").text
+    # Process 'Contents' elements
+    objects = doc.xpath("//xmlns:Contents").map do |node|
+      Blob.from_xml(node, bucket_name, encoding_type: encoding_type)
+    end
+
+    # Note to future: We may need to process 'CommonPrefixes' elements
+    # when we implement new APIs.
+    is_truncated = doc.xpath("//xmlns:IsTruncated").text.casecmp("true").zero?
+    continuation_token = doc.xpath("//xmlns:NextContinuationToken").text
+    if continuation_token && encoding_type == "url"
+      continuation_token = CGI.escape(continuation_token)
+    end
+    [objects, is_truncated, continuation_token]
+  end
+
+  class Blob
+    attr_accessor :bucket_name, :key, :last_modified, :etag, :size,
+      :version_id, :is_latest, :storage_class, :owner_id, :owner_name,
+      :metadata, :is_delete_marker
+
+    def initialize(bucket_name, object_name, last_modified: nil, etag: nil, size: nil,
+      version_id: nil, is_latest: nil, storage_class: nil, owner_id: nil,
+      owner_name: nil, metadata: {}, is_delete_marker: false)
+      @bucket_name = bucket_name
+      @key = object_name
+      @last_modified = last_modified
+      @etag = etag
+      @size = size
+      @version_id = version_id
+      @is_latest = is_latest
+      @storage_class = storage_class
+      @owner_id = owner_id
+      @owner_name = owner_name
+      @metadata = metadata
+      @is_delete_marker = is_delete_marker
+    end
+
+    def self.from_xml(element, bucket_name, encoding_type: nil, is_delete_marker: false)
+      last_modified = element.at_xpath("xmlns:LastModified")&.text
```

Comment:
This should be timestamp, otherwise we diverge from the original response (and all the callers need to parse this separately). Encapsulating it with Time.parse should be enough.